src/database/migrations/0001_create_canonical_fields.sql

```diff
@@ -0,0 +1,13 @@
+CREATE TABLE canonical_fields (
+  id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  label TEXT,
```

Comment:
all of these should be marked NOT NULL

Reply:
Yes, and text can be varchar.

Reply:
Ah right yes (and we did land on using INTEGER in the end, correct?)

Reply:
Sigh. Yes, I suppose, considering the strangeness/hackiness required for 64-bit integers in JavaScript. Packing/type-alignment would be nice too, but with the 8-byte timestamp and a 4-byte id it seems weird to have the created_at precede the id when looking at the tables.

Reply:
If it makes you feel any better I think future versions of EMCAScript have proposals for bigint being built in

Reply:
I guess ECMAScript has bigint but not necessarily JSON, does that sound right?